tests/Doctrine/Tests/ORM/Hydration/ScalarColumnHydratorTest.php

```diff
@@ -0,0 +1,134 @@
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertTrue(is_array($result));
+        $this->assertEquals(0, count($result));
+    }
+
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testSingleColumnEntityQueryWithoutScalarMap(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        // Faked result set
+        $resultSet = [
+            [
+                'u__id' => '1',
+            ],
+            [
+                'u__id' => '2',
+            ],
+        ];
+
+        $stmt = new HydratorMockResult($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertTrue(is_array($result));
+        $this->assertEquals(2, count($result));
+
+        $this->assertEquals(1, $result[0]);
+        $this->assertEquals(2, $result[1]);
+    }
+
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testSingleColumnEntityQueryWithScalarMap(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+        $rsm->addScalarResult('sclr0', 'id');
+        $rsm->addIndexByScalar('sclr0');
+
+        // Faked result set
+        $resultSet = [
+            [
+                'u__id' => '1',
+            ],
+            [
+                'u__id' => '2',
+            ],
+        ];
+
+        $stmt = new HydratorMockResult($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertEquals(2, count($result));
+        $this->assertTrue(is_array($result));
+
+        $this->assertEquals(1, $result[0]);
+        $this->assertEquals(2, $result[1]);
+    }
+
+    /**
+     * Select u.id, u.name from CmsUser u
+     */
+    public function testMultipleColumnEntityQueryThrowsException(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+        $rsm->addFieldResult('u', 'u__name', 'name');
+
+        // Faked result set
+        $resultSet = [
+            [
+                'u__id' => '1',
+                'u__name' => 'romanb',
+            ],
+            [
+                'u__id' => '2',
+                'u__name' => 'jwage',
+            ],
+        ];
+
+        $stmt = new HydratorMockResult($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+
+        $this->expectException(MultipleSelectorsFoundException::class);
+        $this->expectExceptionMessage(sprintf(
+            MultipleSelectorsFoundException::MULTIPLE_SELECTORS_FOUND_EXCEPTION,
+            implode(', ', $rsm->fieldMappings)
```

Comment:
It's a good idea to reduce complexity in tests, otherwise you would need to test your tests :stuck_out_tongue: 
```suggestion
            'id, name'
```

Reply:
I will update 🤪, but I want to see github actions errors so everything can be fixed together.

Reply:
It's starting to look good :slightly_smiling_face:

Reply:
I learned so many things in last two days 🤘🏻 thank you

Reply:
Updated